client.go

```diff
@@ -924,6 +924,55 @@ func (c *Client) MkdirAll(path string) error {
 	return nil
 }
 
+// RemoveAll delete files recursively in the directory and Recursively delete subdirectories.
+// An error will be returned if no file or directory with the specified path exists
+func (c *Client) RemoveAll(path string) error {
+
+	// Get the file/directory information
+	fi, err := c.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if fi.IsDir() {
+		// Delete files recursively in the directory
+		files, err := c.ReadDir(path)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				// Recursively delete subdirectories
+				err = c.RemoveAll(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			} else {
+				// Delete individual files
+				err = c.Remove(path + "/" + file.Name())
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete the empty directory
+		err = c.RemoveDirectory(path)
+		if err != nil {
+			return c.RemoveDirectory(path)
+		}
```

Comment:
You have misunderstood my previous comments. I was not saying to replace only the `return err` with `return c.RemoveDirectory(path)`. I was saying you can replace _the whole block_.

Any code of the pattern:
```
err := doThing(thing)
if err != nil {
	return err
}
return nil
```
can be replaced by the much simpler `return doThing(thing)`.

If `err != nil` it returns `err`, and if `err == nil` it returns `nil`, which is equivalent to returning `err`. So, both code paths are equivalent to `return err`, and so there is no need to split the code paths in the first place.

Then, once the code is `err := doThing(thing); return err` there’s no longer any need to store the return in a temporary variable.

Reply:
We are returning the `error` being returned by `RemoveDirectory`, there’s just no point in assigning it into a temporary variable, since both code paths are returning that error with no further logic performed on it, we can just return it directly.